src/node_api.cc

```diff
@@ -149,14 +174,14 @@ class ThreadSafeFunction : public node::AsyncResource {
 
   napi_status Push(void* data, napi_threadsafe_function_call_mode mode) {
     node::Mutex::ScopedLock lock(this->mutex);
-    uv_thread_t current_thread = uv_thread_self();
+    ThreadID current_thread;
 
     while (queue.size() >= max_queue_size &&
         max_queue_size > 0 &&
         !is_closing) {
       if (mode == napi_tsfn_nonblocking) {
         return napi_queue_full;
-      } else if (uv_thread_equal(&current_thread, &main_thread)) {
+      } else if (current_thread == main_thread) {
```

Comment:
Is it possible to do something like `env_->is_main_thread()` (worker thread's approach)? in which case we can potentially remove the `ThreadID` class as well as the `main_thread` instance variable?

Reply:
@gireeshpunathil No, because this function may be called from a thread that does not have access to an instance of `node::Environment`.